### PR TITLE
Ung 2220/re retrieve device password

### PR DESCRIPTION
**Pull request by lalottacontinua**

### CODE REVIEW

File: src/main/scala/com/ubirch/services/poc/DeviceCreator.scala [hunk 1]
```diff
@@ -82,11 +90,38 @@ class DeviceCreatorImpl @Inject() (conf: Config, aESEncryption: AESEncryption)(i
...
+              StatusAndPW(status, pw)
+            } else {
+              throwError(PocAndStatus(poc, status), s"unexpected size of thing api response array: ${array.length}; ")
+            }
+          case Left(ex: ResponseError[Exception]) =>
+            throwAndLogError(PocAndStatus(poc, status), "retrieving api-config via Thing API failed: ", ex, logger)
+        }
+      })
+
+  private def getBody(poc: Poc, status: PocStatus, tenant: Tenant): String = {
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/DeviceCreator.scala:117]: Shall we add this annotation here as well?
```
@throws[PocCreationError]
```

File: src/main/scala/com/ubirch/services/poc/DeviceCreator.scala [hunk 2]
```diff
@@ -82,11 +90,38 @@ class DeviceCreatorImpl @Inject() (conf: Config, aESEncryption: AESEncryption)(i
...
+    if (poc.deviceGroupId.isEmpty)
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/DeviceCreator.scala:118]: When this error occurs, does it mean this poc would never been completed until we fix the data manually?

Comment (lalottacontinua) [reply src/main/scala/com/ubirch/services/poc/DeviceCreator.scala:121]: yes, I would say so.

File: src/main/scala/com/ubirch/services/poc/DeviceCreator.scala (hunk 1 above)
Comment (lalottacontinua) [reply src/main/scala/com/ubirch/services/poc/DeviceCreator.scala:120]: good idea.

**Pull request merged**